src/pdn/src/PdnGen.tcl

```diff
@@ -3614,13 +3676,69 @@ proc generate_stripes {tag net_name} {
           set template_name [dict get $plan_template $x $y]
           set layer_info [dict get $grid_data straps $lay $template_name]
           set area [list $x $y [expr $x + [dict get $template width]] [expr $y + [dict get $template height]]]
-          generate_upper_metal_mesh_stripes $tag $lay $layer_info $area
+          generate_upper_metal_mesh_stripes $tag $lay $layer_info $area $net_name
         }
       }
     }
   }
 }
 
+proc insert_power_switches {} {
+  variable grid_data
+
+  set db [ord::get_db]
+  set block [ord::get_db_block]
+
+  set psw [$db findMaster [dict get $grid_data switch_cell]]
+  # The selected power switch is double height, and has a central VSS pin, so align with VSS rails
+  set vgnd [[$psw findMTerm VGND] getMPins]
+  set vddg [[$psw findMTerm VDDG] getMPins]
+
+  set rail_lay [lindex [get_rails_layers] 0]
+  set rail_width [dict get $grid_data rails $rail_lay width]
+
+  set strap_lay [lindex [dict get $grid_data straps] 0]
+  set strap_lay_info [dict get $grid_data straps $strap_lay]
+  set strap_width [dict get $strap_lay_info width]
+
+  set area [dict get $grid_data area]
+  set offset [expr [lindex $area 0] + [dict get $strap_lay_info offset]]
+
+  set row_idx 0
+
+  foreach row [$block getRows] {
+    set orient [$row getOrient]
+    set box [$row getBBox]
+    if {$orient == "R0"} {
+      set vss_y [$box yMin]
+      set overlap_y [expr $vss_y - $rail_width / 2]
+      
+      set col_idx 0
+
+      for {set x $offset} {$x < [expr {[lindex $area 2] - [dict get $strap_lay_info width]}]} {set x [expr {[dict get $strap_lay_info pitch] + $x}]} {
+        set overlap_x [expr $x - $strap_width / 2]
+
+        set location [list \
+          [expr $overlap_x - [[$vddg getBBox] xMin]] \
+          [expr $overlap_y - [[$vgnd getBBox] yMin]] \
+        ]
+
+        set inst_name "PSW_${row_idx}_${col_idx}"
+        if {[set inst [$block findInst $inst_name]] == "NULL"} {
+          set inst [odb::dbInst_create $block $psw $inst_name]
+        }
+
+        $inst setOrigin {*}$location
```

Comment:
In the example you provided, some of the switches are overlapping with the tap cells, so you might need a way to control this better?

Reply:
The simplest approach here is to use power switches that have embedded tie cells. If there is an overlap between a tap cell and a power switch, then the tapcell can simple be deleted.

Alternatively, if power switches without well ties are used, then we can move the tapcell to the left or right (whichever is the shortest move). There is more of a potential for this approach to result in DRC rule violations due to tapcells being too far apart.

We could also have a parameter for the max distance between tapcells, and look to double check this spacing after any adjustments have been made.

Do you have a preference amongst these choices, or else a different approach you would prefer?

Reply:
This might be niche, but I think if a power switch with a tap replaces a tapcell, there is an opportunity for DRVs because the tap within the power switch might be in a different relative location than the tap within the tapcell. This would only happen if the tapcell spacing is extremely close to the maximum DRC distance though.

Reply:
If the power switch has a tap cell included, then I think it would be reasonable to swap it out with the existing tapcell (@rovinski  it's entirely possible this causes a DRV, but only if the designer choose to put the tapcells at the extremes ends of distance, but I'm not sure that the right approach is here since OR doesn't track that information). 
Personally, I think honoring what is already there makes sense and moving to the left or right makes more sense to me. Being able to specify a max distance would  also help since that would allow PDNGEN to double check it's work to ensure it honored all available constraints.

Reply:
@minghungumich  this appears to be issue still. There is no avoidance of existing instances.